From 45e1cee06836861ef79ebe2514a19c956ee96b92 Mon Sep 17 00:00:00 2001
From: John Levon <john.levon@joyent.com>
Date: Tue, 30 Jul 2019 15:07:15 +0000
Subject: [PATCH] OS-7685 smartos-live overlay should be merged into
 illumos-joyent Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by:
 Mike Gerdts <mike.gerdts@joyent.com> Approved by: Mike Gerdts
 <mike.gerdts@joyent.com>

---
 ntp/Solaris/ntp.conf                          | 18 ++++++++--
 ...manifest-and-method-and-install-them.patch | 35 +++++++++++--------
 rsyslog/rsyslog.conf                          |  3 ++
 3 files changed, 39 insertions(+), 17 deletions(-)

diff --git a/ntp/Solaris/ntp.conf b/ntp/Solaris/ntp.conf
index 7827e1d..7097aa8 100644
--- a/ntp/Solaris/ntp.conf
+++ b/ntp/Solaris/ntp.conf
@@ -1,4 +1,16 @@
-server 208.80.96.70
 driftfile /var/ntp/ntp.drift
-restrict default noquery nomodify notrap nopeer
-restrict -6 default noquery nomodify notrap nopeer
+logfile /var/log/ntp.log
+
+# Ignore all network traffic by default
+restrict default ignore
+restrict -6 default ignore
+
+# Allow localhost to manage ntpd
+restrict 127.0.0.1
+restrict -6 ::1
+
+# # Allow servers to reply to our queries
+restrict source nomodify noquery notrap
+
+# Time Servers
+pool 0.smartos.pool.ntp.org burst iburst minpoll 4
diff --git a/openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch b/openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch
index 460c7ce..0332335 100644
--- a/openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch
+++ b/openssh/Patches/0022-Add-SMF-manifest-and-method-and-install-them.patch
@@ -216,11 +216,12 @@ new file mode 100644
 index 00000000..e91ed553
 --- /dev/null
 +++ b/smf/method.sh
-@@ -0,0 +1,121 @@
+@@ -0,0 +1,132 @@
 +#!/sbin/sh
 +#
 +# Copyright 2010 Sun Microsystems, Inc.  All rights reserved.
 +# Use is subject to license terms.
++# Copyright 2015 Joyent, Inc.
 +#
 +
 +. /lib/svc/share/ipf_include.sh
@@ -233,7 +234,7 @@ index 00000000..e91ed553
 +
 +# Checks to see if RSA, and DSA host keys are available
 +# if any of these keys are not present, the respective keys are created.
-+create_key()
++create_key_if_configured()
 +{
 +	keypath=$1
 +	keytype=$2
@@ -265,6 +266,16 @@ index 00000000..e91ed553
 +	fi
 +}
 +
++create_host_key()
++{
++	keytype=$1
++	keyname=ssh_host_${keytype}_key
++	if [ -d $SSHKEYDIR ]; then
++		create_key_if_configured $SSHKEYDIR/$keyname $keytype
++	fi
++	create_key_if_configured $SSHDIR/$keyname $keytype
++}
++
 +create_ipf_rules()
 +{
 +	FMRI=$1
@@ -293,10 +304,10 @@ index 00000000..e91ed553
 +case $1 in
 +	# sysidconfig/sys-unconfig arguments (-c and -u)
 +'-c')
-+	create_key $SSHDIR/ssh_host_rsa_key rsa
-+	create_key $SSHDIR/ssh_host_dsa_key dsa
-+	create_key $SSHDIR/ssh_host_ecdsa_key ecdsa
-+	create_key $SSHDIR/ssh_host_ed25519_key ed25519
++	create_host_key rsa
++	create_host_key dsa
++	create_host_key ecdsa
++	create_host_key ed25519
 +	;;
 +
 +'-u')
@@ -317,10 +328,10 @@ index 00000000..e91ed553
 +	# them; sysidconfig is not run in every situation (such as on
 +	# the install media).
 +	#
-+	create_key $SSHKEYDIR/ssh_host_rsa_key rsa
-+	create_key $SSHKEYDIR/ssh_host_dsa_key dsa
-+	create_key $SSHKEYDIR/ssh_host_ecdsa_key ecdsa
-+	create_key $SSHKEYDIR/ssh_host_ed25519_key ed25519
++	create_host_key rsa
++	create_host_key dsa
++	create_host_key ecdsa
++	create_host_key ed25519
 +
 +	/usr/lib/ssh/sshd
 +	;;
@@ -338,7 +349,3 @@ index 00000000..e91ed553
 +esac
 +
 +exit $?
-\ No newline at end of file
--- 
-2.22.0
-
diff --git a/rsyslog/rsyslog.conf b/rsyslog/rsyslog.conf
index 3976c95..cc37dfb 100644
--- a/rsyslog/rsyslog.conf
+++ b/rsyslog/rsyslog.conf
@@ -20,3 +20,6 @@ mail.debug					/var/log/syslog
 
 auth.info					/var/log/auth.log
 mail.info					/var/log/postfix.log
+
+# Include files from /opt/custom/etc/rsyslog.d if it exists
+$IncludeConfig /opt/custom/etc/rsyslog.d/
-- 
2.21.0

